### ISSUE TITLE
[BUG] TimeSeries.slice_n_points_after indexing

### ISSUE THREAD

**Issue by OliverObst** [unit8co/darts#1480]
**Describe the bug**
The TimeSeries.slice_n_points_after interface expects a time step as a start value.
It uses the time step as an index, potentially leading to empty series (when they should have values). 
It also throws an error when the index is used (but the index is outside of the range of the time steps). As a result, the method is not useable.

**To Reproduce**
```
series = TimeSeries.from_values(np.arange(10,20))
a,b = series.split_after(4)
c = b.slice_n_points_after(6,3)
len(c) == 0

```
**Expected behavior**
In the example above, c should be the time series sliced off after 6, containing the values for time step 7, 8, and 9.
The actual behaviour is that c is empty.

**System (please complete the following information):**
 - Python version: 3.10.9
 - darts version: 0.23.0

**Additional context**
slice_n_point_before likely has the same issue

**Comment by hrzn** [comment]
Thanks for flagging! indeed it seems there's something not working with `__getitem__` when using integer slices, we will have to investigate and fix it.

**Comment by hrzn** [comment]
Fixed in https://github.com/unit8co/darts/pull/1482

**Comment by hrzn** [comment]
This is now fixed and released as part of `0.23.1`, available on PyPI
